graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/ValidationErrorInstrumentation.kt

```diff
@@ -0,0 +1,34 @@
+package com.netflix.graphql.dgs.internal
+
+import com.netflix.graphql.dgs.exceptions.DgsException
+import com.netflix.graphql.types.errors.ErrorDetail
+import com.netflix.graphql.types.errors.ErrorType
+import com.netflix.graphql.types.errors.TypedGraphQLError
+import graphql.ExecutionResult
+import graphql.GraphQLError
+import graphql.execution.instrumentation.InstrumentationState
+import graphql.execution.instrumentation.SimplePerformantInstrumentation
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters
+import graphql.validation.ValidationError
+import java.util.concurrent.CompletableFuture
+
+class ValidationErrorInstrumentation : SimplePerformantInstrumentation() {
+    override fun instrumentExecutionResult(executionResult: ExecutionResult?, parameters: InstrumentationExecutionParameters?, state: InstrumentationState?): CompletableFuture<ExecutionResult> {
+        if (executionResult != null && executionResult.errors.isNotEmpty()) {
+            val newExecutionResult = ExecutionResult.newExecutionResult().from(executionResult)
+            val graphqlErrors: MutableList<GraphQLError> = mutableListOf()
+            executionResult.errors.filterIsInstance<ValidationError>().forEach { validationError ->
+                val graphqlError = TypedGraphQLError
+                    .newBuilder()
+                    .errorType(ErrorType.BAD_REQUEST)
+                    .errorDetail(ErrorDetail.Common.FIELD_NOT_FOUND)
```

Comment:
validationError can represent more than just `FIELD_NOT_FOUND` right?

Reply:
example I can think of is syntax errors

Reply:
Yeah it was what is most appropriate in the given enum. I thought about omitting the error detail to avoid confusion. Alternatively, we could set it if only we get fieldUndefined, but leave it empty for invalid argument, for e.g.?